### ISSUE TITLE
Captcha won't show up when load contact form on cmspage

### ISSUE THREAD

**Issue by JelleGe** [StudioForty9/Recaptcha#1]
Loading the contact form on CMS pages with this code:
{{block type="core/template" name="contactForm" form_action="/contacts/index/post" template="contacts/form.phtml"}}

Then the captcha does not show up.

**Comment by studioforty9-dev** [comment]
Hi, did you add `<?php echo $this->getChildHtml('studioforty9.recaptcha.autorender'); ?>` to your form.phtml template?

**Comment by JelleGe** [comment]
Hi,

Yes I did, and that only works on the default Magento URL http://www.domain.com/contacts 

Regards,
Jelle

**Comment by studioforty9-dev** [comment]
Okay, you'll have to do it a bit differently, open up the cms page in Magento and go to the Design Tab and paste the following under 'Layout Update XML':

```
<reference name="content">
  <block type="core/template" name="contactForm" template="contacts/form.phtml">
    <block type="studioforty9_recaptcha/autorender" name="studioforty9.recaptcha.autorender" template="studioforty9/recaptcha/autorender.phtml"/>
  </block>
</reference>
```

Now remove the `{{block type="core/template" name="contactForm" form_action="/contacts/index/post" template="contacts/form.phtml"}}` block from the cms page content.

**Comment by JelleGe** [comment]
Many thanks! :) 

That worked great.

Regards,
Jelle

**Comment by studioforty9-dev** [comment]
You're very welcome, thanks for using our extension :)

**Comment by JelleGe** [comment]
I have one more question.
The captcha block does use background: #f9f9f9;

But we want to change this to #fff;

How is this possible?

Just add the line to our css file does not work

Regards,
Jelle

**Comment by studioforty9-dev** [comment]
Unfortunately, there just isn't a solution for this currently, Google reCAPTCHA only provides theme options: light and dark. As you said, you can't update the css in the iFrame, have you tried targeting the iFrame via jQuery and setting styles there?

**Comment by JelleGe** [comment]
Hi,

We still got a problem.
Because when some does not enter the reCAPTCHA, the Magento error is not displayed.
Only the page refresh.

I got this problem with this code:

<reference name="content">
  <block type="core/template" name="contactForm" template="contacts/form.phtml">
    <block type="studioforty9_recaptcha/autorender" name="studioforty9.recaptcha.autorender" template="studioforty9/recaptcha/autorender.phtml"/>
  </block>
</reference>

Thanks,
Jelle

**Comment by studioforty9-dev** [comment]
I'm sorry, my bad, I forgot you need to add the "form_action" URL:

```
<reference name="content">
  <block type="core/template" name="contactForm" template="contacts/form.phtml">
    <action method="setFormAction"><value>/contacts/index/post</value></action>
    <block type="studioforty9_recaptcha/autorender" name="studioforty9.recaptcha.autorender" template="studioforty9/recaptcha/autorender.phtml"/>
  </block>
</reference>
```

**Comment by JelleGe** [comment]
Great, many thanks! :) 

It works great now.

Regards,
Jelle

**Comment by JelleGe** [comment]
Hi,

We use this code in the page layout on the CMS page, to load a custom contact form for suggestions: 

<reference name="content">
  <block type="core/template" name="contactForm" template="contacts/suggesties.phtml">
    <action method="setFormAction"><value>/contacts/index/post</value></action>
    <block type="studioforty9_recaptcha/autorender" name="studioforty9.recaptcha.autorender" template="studioforty9/recaptcha/autorender.phtml"/>
  </block>
</reference>

But when the user does not enter the captcha, the Magento error is displayed, but the page redirects to the default Magento contact form.
How can we disable this redirect or edit the redirect to the correct page for this form?

Regards,
Jelle

**Comment by studioforty9-dev** [comment]
Hi,

I've just added some code to make it smarter about where to redirect to, it will now attempt to get the `request_uri` from `visitor_data` in the session, if it can't find that it will fall back to the last url and finally if it can't find that it will use `/contacts`.

I've tested on my machine and it works as expected, can you test it out on your site and let me know how you get on?

Thanks!

**Comment by JelleGe** [comment]
Hi,

Great, good to hear! :)
Where can I find the latest version?

**Comment by eoghanobrien** [comment]
dev-master - the most recent commit will have it.

**Comment by JelleGe** [comment]
Do you also make it available to the magento connect manager?

**Comment by eoghanobrien** [comment]
No, not until it's tested more.

**Comment by JelleGe** [comment]
Hi,

I found one bug.
When you ignore the captacha, the message appear with something like “You should use the captcha”.
But the message of Magento is displayed twice. (see screenshot).

**Comment by JelleGe** [comment]
Sorry my mistake, I did echo the line again in the template! :)

It works great now :)

**Comment by JelleGe** [comment]
Hi,

There is only one thing, in your script you echo the Magento error message.
So I need to delete this line in my template, but you do not echo the Magento correct message. 

So when the form is sent correctly, people do not know that because this line is not displayed.

**Comment by eoghanobrien** [comment]
Hi Jelle,

In which file are you seeing the 'echo'?

**Comment by JelleGe** [comment]
Well your extention does load the default Magento messages.
But only the error message does load, the correct sent message is not displayed.

Jelle

**Comment by sarrafj** [comment]
it is not showing in the easytabs. i am using a custom theme and when i check out the recaptcha from the review page url ie : mywebsite.com/review/product/list/id/73/category/50/
the captcha loads perfectly on the review page.  but on the product view page under the easytabs there is only the review form but no recaptcha. i have also tried calling the code on local.xml file.. nothing is changed. help me ..

**Comment by studioforty9-dev** [comment]
Sorry sarrafj, your issue is related to a custom theme rather than displaying the widget on the supported pages. 

Please start another issue thread and I'll see what I can do to help, most likely you need to find the correct place in you layout xml to add the review form.
